src/crimson/osd/pg.cc

```diff
@@ -847,11 +866,14 @@ std::optional<hobject_t> PG::resolve_oid(
   const SnapSet &ss,
   const hobject_t &oid)
 {
+  logger().debug("{} oid.snap={},head snapset.seq={}",
+                 __func__, oid.snap, ss.seq);
   if (oid.snap > ss.seq) {
+    // perfrom a read from head
     return oid.get_head();
   } else {
     // which clone would it be?
-    auto clone = std::upper_bound(
+    auto clone = std::find(
```

Comment:
why? if we want to ensure that `clone` is of the same `oid.snap`, shall we use `std::lower_bound()`? do you want to optimize for the case of small `std::size(ss.clones)`?

Reply:
Changed to `lower_bound()` for now. This optimization seems to be negligible.